sdk/metric/internal/sum.go

```diff
@@ -35,21 +48,32 @@ func (s *sum[N]) Aggregate(value N, attr attribute.Set) {
 // Each aggregation cycle is treated independently. When the returned
 // Aggregator's Aggregations method is called it will reset all sums to zero.
 func NewDeltaSum[N int64 | float64]() Aggregator[N] {
-	// TODO(#2972): implement.
-	return &deltaSum[N]{}
+	return &deltaSum[N]{newSum[N]()}
 }
 
 // deltaSum summarizes a set of measurements made in a single aggregation
 // cycle as their arithmetic sum.
 type deltaSum[N int64 | float64] struct {
 	sum[N]
-
-	// TODO(#2972): implement.
 }
 
 func (s *deltaSum[N]) Aggregations() []Aggregation {
-	// TODO(#2972): implement.
-	return nil
+	now := time.Now().UnixNano()
+
+	s.Lock()
+	defer s.Unlock()
+
+	aggs := make([]Aggregation, 0, len(s.values))
+	for attr, value := range s.values {
+		aggs = append(aggs, Aggregation{
+			Timestamp:  now,
+			Attributes: attr,
+			Value:      SingleValue[N]{Value: value},
+		})
+		delete(s.values, attr)
```

Comment:
I'm not sure if this should stop tracking the attribute set or just reset it to some zero.

I'm going to look at other implementations to better understand the landscape.

Reply:
Looks like python resets to zero: https://github.com/open-telemetry/opentelemetry-python/blob/25771ecdac685a5bf7ada1da21092d2061dbfc02/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py#L134

Reply:
Java also looks to reset to zero: https://github.com/open-telemetry/opentelemetry-java/blob/aa873a05be27137d276de5fa50557add5f78184c/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/LongSumAggregator.java#L98, https://github.com/open-telemetry/opentelemetry-java/blob/aa873a05be27137d276de5fa50557add5f78184c/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/concurrent/AtomicLongLongAdder.java#L36.

Reply:
Updating to follow suit.

Reply:
[This example](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#synchronous-example-delta-aggregation-temporality) indicates that the set should indeed be deleted.